src/org/nuxeo/openshift/library/ConnectClient.groovy

```diff
@@ -0,0 +1,87 @@
+package org.nuxeo.openshift.library
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import java.util.ArrayList
+import groovy.lang.Closure
+import org.apache.http.auth.UsernamePasswordCredentials
+import org.apache.http.client.methods.HttpPost
+import org.apache.http.entity.StringEntity
+import org.apache.http.impl.client.BasicCredentialsProvider
+import org.apache.http.impl.client.CloseableHttpClient
+import org.apache.http.impl.client.HttpClientBuilder
+
+import static org.apache.http.auth.AuthScope.ANY
+
+def deployPackageToConnect(connectUsername, connectPassword, studioProject, packageFilePath, connectUrl='https://connect.nuxeo.com') {
```

Comment:
I thought that this method did not work with Token authentication?

Reply:
Correct
That's why I wrote:
https://github.com/nuxeo/nuxeo-openshift/blob/b8285a07282971e36aa3dc4b4c4cd62ab346fa09/jenkinsfile/Jenkinsfile-release#L188